### PR TITLE
Apply black-style formatting to repository

### PR DESCRIPTION
**Pull request by j6k4m8**
@JeanKossaifi — this is a follow-up PR to #400 and #399.

There are pretty extensive changes here but I have spot-checked and I think that this does indeed tend to improve readability. 

The linting here was all performed by running `black .` in the root directory of this repository. If you like the changes, I will also add instructions for how to install and run `black` to the documentation for new contributors!

### CODE REVIEW

**Comment by JeanKossaifi** [comment]
Looks like we've messed up with the history while merging other PR - there seem to be a few conflicts.

**Comment by j6k4m8** [comment]
No prob, fixed!

**Review by JeanKossaifi — COMMENTED** [review]
Thanks @j6k4m8! Can we exclude sphinx_ext as these are just external plugins? 
I realize this is huge to review - would it be easy to split into smaller PRs, e.g. per submodule so we can review quickly? Otherwise I can just carefully go through each file.

**Comment by JeanKossaifi** [comment]
Just wanted to check whether you have time to look at this @j6k4m8 :) 
Looking forward to have it merged!

**Comment by j6k4m8** [comment]
This is funny timing! I was just thinking about this PR yesterday.

I wonder if it might make sense to NOT include file changes in this PR, and let people update files as they are committed? We could use something like [this](https://github.com/marketplace/actions/python-black-formatter-only-on-edited-files). That way we never have a huge change to the codebase all at once.

Alternatively, I'm happy to fix these merge conflicts and edit to match your suggestions, and get this merged in asap :) I think it should be a pretty quick fix.

**Comment by JeanKossaifi** [comment]
Thanks @j6k4m8!

I would actually go with both: fix formatting in this PR and update the workflow to only run Black on the updated files.
What do others think @tensorly/tensorly?

**Comment by aarmey** [comment]
I'd rather merge this to get all of the linting noise out of the way in one PR. If we do it file-by-file, many PRs will have extra linting noise.

**Comment by aarmey** [comment]
I'm happy to help review once the merge conflicts are dealt with.

File: examples/regression/plot_cp_regression.py
Comment (j6k4m8) [reply examples/regression/plot_cp_regression.py:61]: Unfortunately this is still an outstanding problem in black (https://github.com/psf/black/issues/148), my current fix is to add `# fmt: skip` on these lines for now. Thoughts welcome!

**Comment by j6k4m8** [comment]
@aarmey and @JeanKossaifi — I think this is ready for your review again! 

One thing to keep in mind is that Black is highly opinionated and does not really have any sort of configurability if you disagree with its rules. On the other hand, this makes things super consistent between codebases, which is awesome. But if you two tend to disagree with its stylistic changes, then we could consider using another linter like pylint or yapf! It adds a bit more config complexity, but it's not the end of the world.

**Review by aarmey — APPROVED** [review]
I've looked through everything and don't see any issues. There are certainly places where I'd make a different decision about formatting, but this is a huge improvement in readability and black's opinionated formatting will help navigate diffs.

**Comment by JeanKossaifi** [comment]
My thinking is that if we later go through individual files and override Black, it may be good to have the checks limited to the changed files (assuming we choose to just ignore the warning and not explicitly skip in the code). What do you think?

**Comment by j6k4m8** [comment]
I'm happy either way — although I think it may be unnecessarily complicated :)

**Comment by aarmey** [comment]
Perhaps we can merge this and then figure out the black setup after? @j6k4m8 would you be able to take care of the merge conflicts? If so I can then re-review.

**Comment by JeanKossaifi** [comment]
Yes, let's merge this PR first before moving to others to avoid more merge conflicts.

I'm fine with merging as is and making the change to the worflow in another PR. I would prefer to only apply Black to modified files to make the PRs clearer to read and focused on the actual changes, esp. if we decide to purposefully deviate from Black in some specific files.

**Review by aarmey — APPROVED** [review]
Looks good to me! I'll wait until the tests pass to merge.

**Review by JeanKossaifi — APPROVED** [review]
Looks good to me too, thanks @j6k4m8!

**Pull request merged**
